imx_usb.c

```diff
@@ -505,55 +487,48 @@ int main(int argc, char * const argv[])
 		goto out_close_usb;
 	}
 
-	// By default, use work from config file...
-	curr = p_id->work;
-
-	if (cmd_head != NULL)
-		curr = cmd_head;
-
-	if (curr == NULL) {
-		fprintf(stderr, "no job found\n"); 
-		ret = EXIT_FAILURE;
-		goto out_close_usb;
-	}
-
 	while (curr) {
+		/* Do current job */
 		if (curr->mem)
 			perform_mem_work(p_id, curr->mem);
-//		printf("jump_mode %x\n", curr->jump_mode);
-		if (curr->filename[0]) {
+		if (curr->filename[0])
 			err = DoIRomDownload(p_id, curr, verify);
-		}
 		if (err) {
 			err = do_status(p_id);
 			break;
 		}
+
+		/* Check if more work is to do... */
 		if (!curr->next && (!curr->plug || curr != cmd_head))
-			break;
+			curr = NULL;
```

Comment:
Is this part working? Here you remove the break and set curr to NULL. Then there are operations done on curr. It seems that next patches fix that but still it'd be best if the history can be "bisectable".